### DISCUSSION TITLE
How to update a one-off-container from a docker-compose file

### DISCUSSION THREAD

**Discussion by ace130-github** [containrrr/watchtower#1090] [Q&A]
We have a liquibase-type container within a docker-compose file. This container runs once, updates the database to the latest schema, and then exits. We have this container updated by Watchtower with options `--include-stopped` and `--revive-stopped`. We can see that Watchtower correctly performs the steps: detect newer images, pull them, and run the container. However, the container does not run off the newer image, but off the previous image, therefore **not** updating the database.

Our suspicion is that Watchtower does the equivalent of `docker-compose pull` & `docker-compose start` when reviving a stopped container. But this starts the old container, not using the newer image it had just pulled. 

What would be needed in this situation would be the equivalent of `docker-compose pull` & `docker-compose rm` & `docker-compose start`.

Is there a possibility to configure Watchtower in this way? If not, how can a Watchtower update of such a short-running container be handled?

**Comment by piksel** [comment]
Watchtower doesn't do anything compose-specific, but it *does* recreate the container with the new image, so this *should* work the way you want. You can run it with `--debug` to see a bit more details in the log.

**Comment by ace130-github** [comment]
Thank you @piksel for your swift reply. Will try again with `--debug` to see where the difference comes from. As far as I can see today this works as described for containers in running state, but not for containers that have run to completion, and are started again via `--revive-stopped`.

**Reply by piksel** [reply]
Yeah, there might be a bug there, I have never tried it, but from the code path it looks like it's doing the right thing...
